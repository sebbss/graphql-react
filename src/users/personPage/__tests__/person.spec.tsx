import React from 'react';
import TestRenderer from 'react-test-renderer'
import { BrowserRouter as Router} from 'react-router-dom'
import PersonPage from '../component';
import { ApolloError } from '@apollo/client';
import { Spinner } from 'baseui/icon';

describe("PersonPage should render correctly",()=>{
    it("should render without crushing", ()=>{
        const component = TestRenderer.create(
            <Router>
                <PersonPage
                data={undefined}
                error={new ApolloError({networkError: new Error("error message")})}
                loading={false}
                />
            </Router>
        )
        expect(component.toJSON()).toMatchSnapshot()
        expect(component.toJSON()).toBeDefined()
    })

    it("should show loader", async ()=>{
        const component = TestRenderer.create(
            <Router>
                <PersonPage
                data={undefined}
                error={new ApolloError({networkError: new Error("error message")})}
                loading={true}
                />
            </Router>
        )
        const loader = await component.root.findByType(Spinner)
        expect(loader).toBeDefined()
    })
})
