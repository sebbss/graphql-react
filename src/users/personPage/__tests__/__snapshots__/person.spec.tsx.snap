// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonPage should render correctly should render without crushing 1`] = `
<div
  className=""
>
  <button
    className=""
    data-baseweb="button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
  >
    <div
      className=""
    >
      <svg
        className=""
        data-baseweb="icon"
        title="Arrow Left"
        viewBox="0 0 24 24"
      >
        <title>
          Arrow Left
        </title>
        <path
          clipRule="evenodd"
          d="M6.29289 11.2929C5.90237 11.6834 5.90237 12.3166 6.29289 12.7071L10.2929 16.7071C10.6834 17.0976 11.3166 17.0976 11.7071 16.7071C12.0976 16.3166 12.0976 15.6834 11.7071 15.2929L9.41421 13H17C17.5523 13 18 12.5523 18 12C18 11.4477 17.5523 11 17 11H9.41421L11.7071 8.70711C12.0976 8.31658 12.0976 7.68342 11.7071 7.29289C11.3166 6.90237 10.6834 6.90237 10.2929 7.29289L6.29289 11.2929Z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    Back
  </button>
  <div
    className=""
    data-baseweb="notification"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="alert"
  >
    <div
      className=""
    >
      error message
    </div>
    <svg
      className=""
      data-baseweb="icon"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
      title="Close"
      viewBox="0 0 24 24"
    >
      <title>
        Close
      </title>
      <path
        clipRule="evenodd"
        d="M7.29289 7.29289C7.68342 6.90237 8.31658 6.90237 8.70711 7.29289L12 10.5858L15.2929 7.29289C15.6834 6.90237 16.3166 6.90237 16.7071 7.29289C17.0976 7.68342 17.0976 8.31658 16.7071 8.70711L13.4142 12L16.7071 15.2929C17.0976 15.6834 17.0976 16.3166 16.7071 16.7071C16.3166 17.0976 15.6834 17.0976 15.2929 16.7071L12 13.4142L8.70711 16.7071C8.31658 17.0976 7.68342 17.0976 7.29289 16.7071C6.90237 16.3166 6.90237 15.6834 7.29289 15.2929L10.5858 12L7.29289 8.70711C6.90237 8.31658 6.90237 7.68342 7.29289 7.29289Z"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className=""
  >
    <section
      className=""
      data-baseweb="card"
    >
      <div
        className=""
      >
        <div
          className=""
        >
          <div
            className=""
          >
            <p>
               Height: 
              undefined meters
               
            </p>
            <p>
              Gender: 
            </p>
            <p>
              Mass: 
            </p>
            <p>
              Homeworld: 
            </p>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
